src/librustc_mir/tcx/pattern.rs

```diff
@@ -153,38 +151,38 @@ impl<'tcx> Mirror<'tcx> for PatNode<'tcx> {
                 let hi = const_eval::eval_const_expr(cx.tcx, hi);
                 let hi = Literal::Value { value: hi };
                 PatternKind::Range { lo: lo, hi: hi }
-            },
+            }
 
             hir::PatEnum(..) | hir::PatIdent(..) | hir::PatQPath(..)
-                if pat_is_resolved_const(&cx.tcx.def_map, self.pat) =>
-            {
+                if pat_is_resolved_const(&cx.tcx.def_map, self.pat) => {
                 let def = cx.tcx.def_map.borrow().get(&self.pat.id).unwrap().full_def();
```

Comment:
I find this problematic (and honestly I have trouble deciding how to format it myself). In particular, the `if` here has the same indentation as the arm body, and I find them visually very hard to distinguish. This is why I moved the `{` to the next line, though I don't love that solution... but I do think we should try to do SOMETHING.